src/Lexer.js

```diff
@@ -219,9 +225,15 @@ module.exports = class Lexer {
       }
 
       // text
-      if (token = this.tokenizer.text(src)) {
+      if (token = this.tokenizer.text(src, tokens)) {
         src = src.substring(token.raw.length);
-        tokens.push(token);
+        if (token.type) {
```

Comment:
DRY? Same code added in two places?

Reply:
That code is not really related so they could change independently.

Reply:
I don't think we should combine that code into a function since it could change and need to be moved back anyway. This is one reason the best practice for DRY is to only combine code that needs to change together and only if it is in three or more places. [Rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming))